lib/utils.js

```diff
@@ -32,3 +32,81 @@ exports.hasGlobstarEnding = function(path) {
 exports.replaceTech = function(path, value) {
     return PATH.join(PATH.dirname(path), PATH.basename(path).split('.').shift()) + '.' + value;
 };
+
+/**
+ * Returns list of files in path indicated by guard value:
+ *  - true indicates that the entry should be included
+ *  - false indicates that the file should be excluded, but should still be traversed if it is a directory
+ *  - null indiciates that a directory should not be traversed
+ *
+ * @param {String} path
+ * @param {GuardFunction} guard
+ * @returns {Promise<String[]>}
+ */
+exports.listTree = (path, guard) => {
+    return new Promise((resolve, reject) => {
```

Comment:
Prob better to use something like https://github.com/normalize/mz or https://github.com/sindresorhus/pify